From 1e36c0bc40edd92db56346bc00b23978a87c602b Mon Sep 17 00:00:00 2001
From: Frank van der Linden <fllinden@amazon.com>
Date: Sun, 24 Mar 2019 22:41:38 +0000
Subject: lustre: adapt for fault / page_mkwrite return type change

The return type for the fault and page_mkwrite handlers
was changed to vm_fault_t - adapt.

Linux commit: 3d3539018d2cbd12e5af4a132636ee7fd8d43ef0 ("mm: create the new vm_fault_t type")

Signed-off-by: Frank van der Linden <fllinden@amazon.com>
---
 drivers/staging/lustrefsx/lustre/include/lustre_compat.h | 8 ++++++++
 drivers/staging/lustrefsx/lustre/llite/llite_mmap.c      | 4 ++--
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/lustrefsx/lustre/include/lustre_compat.h b/drivers/staging/lustrefsx/lustre/include/lustre_compat.h
index 115d7f214148..2fb308a998ac 100644
--- a/drivers/staging/lustrefsx/lustre/include/lustre_compat.h
+++ b/drivers/staging/lustrefsx/lustre/include/lustre_compat.h
@@ -730,4 +730,12 @@ static inline void unlock_mappings(struct address_space *mappings)
 #define TOTALRAM_PAGES totalram_pages
 #endif
 
+#ifdef HAVE_VM_OPS_USE_VM_FAULT_ONLY
+# ifdef HAVE_VM_FAULT_T
+#  define VM_FAULT_T vm_fault_t
+# else
+#  define VM_FAULT_T int
+# endif
+#endif
+
 #endif /* _LUSTRE_COMPAT_H */
diff --git a/drivers/staging/lustrefsx/lustre/llite/llite_mmap.c b/drivers/staging/lustrefsx/lustre/llite/llite_mmap.c
index 0fff9b9663a9..79adfa937851 100644
--- a/drivers/staging/lustrefsx/lustre/llite/llite_mmap.c
+++ b/drivers/staging/lustrefsx/lustre/llite/llite_mmap.c
@@ -333,7 +333,7 @@ static int ll_fault0(struct vm_area_struct *vma, struct vm_fault *vmf)
 }
 
 #ifdef HAVE_VM_OPS_USE_VM_FAULT_ONLY
-static int ll_fault(struct vm_fault *vmf)
+static VM_FAULT_T ll_fault(struct vm_fault *vmf)
 {
 	struct vm_area_struct *vma = vmf->vma;
 #else
@@ -382,7 +382,7 @@ static int ll_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 }
 
 #ifdef HAVE_VM_OPS_USE_VM_FAULT_ONLY
-static int ll_page_mkwrite(struct vm_fault *vmf)
+static VM_FAULT_T ll_page_mkwrite(struct vm_fault *vmf)
 {
 	struct vm_area_struct *vma = vmf->vma;
 #else
-- 
2.32.0

